File: WinFormsApp1/Form1.vb
Imports System.ComponentModel
Imports System.Runtime.InteropServices
Imports Vanara.PInvoke
Imports Vanara.PInvoke.CldApi

Public Class Form1
    Private VirtualizationRootPath As String
    Private _SyncProvider As SyncProvider


    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim source As String = "D:\TEMP"
        Me.VirtualizationRootPath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + "\VirtualTest3"

        If Not IO.Directory.Exists(source) Then IO.Directory.CreateDirectory(source)

        Me._SyncProvider = New SyncProvider(Me.VirtualizationRootPath, New FolderProvider(source))
        Me._SyncProvider.Start()

        Me.Button1.Enabled = False
        Me.Button2.Enabled = True
        Me.Button3.Enabled = True
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me._SyncProvider.Stop()

        Me.Button1.Enabled = True
        Me.Button2.Enabled = False
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Me._SyncProvider.DeleteLocalData()

        Me.Button1.Enabled = True
        Me.Button2.Enabled = False
        Me.Button3.Enabled = False
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Me._SyncProvider.Unregister()
    End Sub
End Class

File: ClassLibrary1/Form1.cs
using Styletronix.CloudSyncProvider;
using System;
using System.Windows.Forms;

namespace ClassLibrary1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private SyncProvider SyncProvider;

        private void Form1_Load(object sender, EventArgs e)
        {
            var param = new SyncProviderParameters()
            {
                ProviderInfo = new BasicSyncProviderInfo()
                {
                    ProviderId = Guid.Parse(@"fdf0b5bb-be08-4544-b6f6-fa954e869a87"),
                    CLSID = @"{DBD02E04-87CD-4E46-A230-FCE0B7B70FEB}",
                    ProviderName = "SXTestProvider",
                    ProviderVersion = "0.0.1"
                },
                LocalDataPath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + @"\VirtualTest3",
                ServerProvider = new ServerProvider(@"D:\TEMP")
            };

            SyncProvider = new SyncProvider(param);
        }

        private async void Button1_Click(object sender, EventArgs e)
        {
            this.button1.Enabled = false;
            await SyncProvider.Start();
        }
        private async void Button5_Click(object sender, EventArgs e)
        {
            this.button5.Enabled = false;
            await SyncProvider.Stop();
        }

        private async void Button2_Click(object sender, EventArgs e)
        {
            this.button2.Enabled = false;
            await SyncProvider.RevertAllPlaceholders();
        }

        private async void Button3_Click(object sender, EventArgs e)
        {
            this.button3.Enabled = false;
            await SyncProvider.Unregister();
        }

        private async void Button4_Click(object sender, EventArgs e)
        {
            await SyncProvider.SyncDataAsync(SyncProvider.SyncMode.Local);
        }

        private async void Button6_Click(object sender, EventArgs e)
        {
            await SyncProvider.SyncDataAsync(SyncProvider.SyncMode.Full);
        }
    }
}
